### PR TITLE
cartage-s3 2.0

### PR DESCRIPTION
**Pull request by halostatue**
-   Rewrote for compatibility with cartage 2.0.
-   3 major enhancements
  -   Configuration now supports multiple destinations. Old cartage-s3
    configurations will continue to work and be accessible under the
    destination name +default+. An error will be raised if there is an
    explicit +default+ destination combined with this implicit +default+
    destination.
  -   The <tt>s3 put</tt> command (<tt>cartage s3 put</tt>) does _not_ cause
    the build of a package, but will only upload a build file. This
    requires that a timestamp be provided either on the command-line or
    through a configuration file.
    
    ```
    cartage --timestamp=<TIMESTAMP> s3 put
    ```
  -   Added <tt>s3 get</tt>, <tt>s3 list</tt>, and <tt>s3 delete</tt>
    commands.

Added more functionality

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,9 +1,16 @@
...
 
-source "https://rubygems.org/"
+source 'https://rubygems.org/'
+
+# Specify your gem's dependencies in cartage-bundler.gemspec
 gemspec
 
-# vim: syntax=ruby
+group :local_development, :test do
+  gem 'cartage', path: '../cartage'
```
Comment (ravster) [inline Gemfile:12]: Do we need `path:` in here?

Comment (halostatue) [inline Gemfile:12]: That’s for local development and only applies when `$LOCAL_DEV` is set.

File: test/test_cartage_s3.rb [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+  }
+
+  describe '#put' do
+    it_verifies_configuration do
+      instance_stub subject.class, :put_file, ->(*) {} do
+        subject.put
+      end
+    end
+
+    it '#put uploads the release metadata and packages' do
```
Comment (ravster) [inline test/test_cartage_s3.rb]: remove `#put`?

**Comment by llwire** [comment]
+1

**Comment by llwire** [comment]
+1

**Pull request merged**